cmd/kyma/alpha/disable/module/module.go

```diff
@@ -0,0 +1,181 @@
+		return errors.New("you must pass one Kyma module name to disable")
+	}
+	moduleName := args[0]
+
+	if err := cmd.opts.validateFlags(); err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, cmd.opts.Timeout)
+	defer cancel()
+
+	return cmd.run(ctx, moduleName)
+}
+
+func (cmd *command) run(ctx context.Context, moduleName string) error {
+	start := time.Now()
+
+	if cmd.K8s == nil {
+		var err error
+		if cmd.K8s, err = kube.NewFromConfigWithTimeout("", cmd.KubeconfigPath, cmd.opts.Timeout); err != nil {
+			return fmt.Errorf("failed to initialize the Kubernetes client from given kubeconfig: %w", err)
+		}
+	}
+
+	if _, err := clusterinfo.Discover(ctx, cmd.K8s.Static()); err != nil {
+		return err
+	}
+
+	kyma, err := cmd.K8s.Dynamic().Resource(kymaResource).Namespace(cmd.opts.Namespace).Get(
+		ctx, cmd.opts.KymaName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not get Kyma %s/%s: %w", cmd.opts.Namespace, cmd.opts.KymaName, err)
+	}
+
+	modules, _, err := unstructured.NestedSlice(kyma.UnstructuredContent(), "spec", "modules")
+	if err != nil {
+		return fmt.Errorf("could not parse modules spec: %w", err)
+	}
+
+	desiredModules, err := disableModule(modules, moduleName, cmd.opts.Channel)
+	if err != nil {
+		return fmt.Errorf("could not enable module: %w", err)
+	}
+
+	if len(modules) != len(desiredModules) {
+		err = unstructured.SetNestedSlice(kyma.Object, desiredModules, "spec", "modules")
+		if err != nil {
+			return fmt.Errorf("failed to set modules list in Kyma spec: %w", err)
+		}
+		_, err = cmd.K8s.Dynamic().Resource(kymaResource).Namespace(cmd.opts.Namespace).Update(
+			ctx, kyma, metav1.UpdateOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to update Kyma %s in %s: %w", cmd.opts.KymaName, cmd.opts.Namespace, err)
+		}
+
+		if cmd.opts.Wait {
+			time.Sleep(2 * time.Second)
+			checkFn := func(u *unstructured.Unstructured) (bool, error) {
+				status, exists, err := unstructured.NestedString(u.Object, "status", "state")
+				if err != nil {
+					return false, errors.Wrap(err, "error waiting for Kyma readiness")
+				}
+				return exists && status == "Ready", nil
+			}
+			err = cmd.K8s.WatchResource(kymaResource, cmd.opts.KymaName, cmd.opts.Namespace, checkFn)
+			if err != nil {
+				return errors.Wrap(err, "failed to watch resource Kyma for state 'Ready'")
+			}
+		}
+	}
+
+	l := cli.NewLogger(cmd.opts.Verbose).Sugar()
+	l.Infof("disabling module took %s", time.Since(start))
+
+	return nil
+}
+
+func disableModule(modules []interface{}, name, channel string) ([]interface{}, error) {
+	for i := range modules {
+		module, _ := modules[i].(map[string]interface{})
+		moduleName, found := module["name"]
+		if !found {
+			return nil, errors.New("invalid item in modules spec: name field missing")
+		}
+		if moduleName == name {
+			moduleChannel, cFound := module["channel"]
+			if cFound && moduleChannel != channel {
+				continue
+			}
+			copy(modules[i:], modules[i+1:])
```

Comment:
Is the logic here just trying to remove one entry in the slice? Then I suggest change to [this way](https://stackoverflow.com/a/37335777), just record the index first, then append the slice before and after.

https://stackoverflow.com/a/37335777